lib/install/install.go

```diff
@@ -103,13 +104,26 @@ func (i *Installer) Execute(req *installpb.ExecuteRequest, stream installpb.Agen
 	if !i.submit(req) && !req.IsResume() {
 		return status.Error(codes.AlreadyExists, "operation is already active")
 	}
-	for event := range i.dispatcher.Chan() {
-		err := stream.Send(event)
-		if err != nil {
-			return trace.Wrap(err)
+	for {
+		select {
+		case event := <-i.dispatcher.Chan():
+			err := stream.Send(event)
+			if err != nil {
+				return trace.Wrap(err)
+			}
+		case result := <-i.execDoneC:
```

Comment:
This implements the logic of handling completion of a single `operation step` (be it the complete install operation or a single phase in case the installer is resumed). The difficulty here is that this case needs to handle client which might temporarily drop and not be able to accept the notification. It also handles the sending of the final completion event directly to client (bypassing the dispatcher which is used for other but non-critical notifications).